eogrow/tasks/spatial.py

```diff
@@ -0,0 +1,248 @@
+            raise ValueError("EOPatch is missing a bounding box")
+        if bbox is not None:
+            if main_bbox.crs is not bbox.crs:
+                raise ValueError("Given bbox is not in the same CRS as EOPatch bbox")
+            if not main_bbox.geometry.contains(bbox.geometry):
+                raise ValueError("Given bbox must be fully contained in EOPatch's bbox")
+
+        sliced_eopatch = EOPatch(bbox=bbox)
+
+        for feature in self.features:
+            feature_type, _ = feature
+            if feature_type is FeatureType.BBOX:
+                continue
+            data = eopatch[feature]
+
+            if feature_type.is_spatial():
+                if bbox is None:
+                    continue
+
+                if feature_type.is_raster():
+                    sliced_data = self._slice_raster(data, main_bbox, bbox)
+                else:
+                    sliced_data = self._filter_vector(data, bbox)
+            else:
+                sliced_data = data
+
+            sliced_eopatch[feature] = sliced_data
+
+        return sliced_eopatch
+
+
+def get_array_slices(
+    bbox: BBox,
+    slice_bbox: BBox,
+    *,
+    resolution: Optional[Union[np.ndarray, Tuple[float, float]]] = None,
+    size: Optional[Union[np.ndarray, Tuple[int, int]]] = None,
+    raise_misaligned: bool = True,
+    limit_x: Optional[Tuple[int, int]] = None,
+    limit_y: Optional[Tuple[int, int]] = None,
+) -> Tuple[slice, slice]:
+    """Slicing taken from eolearn.io.ImportFromTiffTask.
+
+    :param bbox: A bounding box of initial array.
+    :param slice_bbox: A bounding box of array to be sliced.
+    :param resolution: A working resolution.
+    :param size: A working size.
+    :param raise_misaligned: Whether to raise an error if the slice would be pixel misaligned the initial array.
+    :param limit_x: If provided it will clip the horizontal slice to a given interval.
+    :param limit_y: If provided it will clip the vertical slice to a given interval.
+    :return: A slice over horizontal direction and a slice over vertical direction.
+    """
+    raster_upper_left = np.array([bbox.min_x, bbox.max_y])
+
+    if size is not None and resolution is None:
+        width, height = size
+        raster_lower_right = np.array([bbox.max_x, bbox.min_y])
+        resolution = np.abs((raster_upper_left - raster_lower_right)) / (width, height)
+    elif size is None and resolution is not None:
+        resolution = np.array(resolution)
+    else:
+        raise ValueError("Only one of the resolution and size can be given")
+
+    slice_upper_left = np.array([slice_bbox.min_x, slice_bbox.max_y])
+    slice_lower_right = np.array([slice_bbox.max_x, slice_bbox.min_y])
+    axis_flip = [1, -1]  # image origin is upper left, geographic origin is lower left
+
+    offset = axis_flip * (slice_upper_left - raster_upper_left) / resolution
+    try:
+        offset_x, offset_y = convert_to_int(offset, raise_diff=raise_misaligned)
```

Comment:
because `convert_to_int` works on `np.ndarrays` the types are set to `Any` after unpacking. Just a warning that annotations should be added if one wants to have more safety when using the function. Also is it ever used on large arrays? so far i've only seen it used on small arrays, and maybe returning `Tuple[int, ...]` would also be ok. Just food for tought

Reply:
I think it would be weird if `convert_to_int` would besides converting values to ints also convert a numpy array to tuple. Perhaps this issue could be addressed again when we add better numpy types altogether?